fix: sanitizers false positives

kostasrim opened the pull request:
Fixes #3443

This is a known issue, see:

1. https://github.com/google/sanitizers/issues/1760
2. https://github.com/dragonflydb/dragonfly/pull/3176
I don't use, `ASAN_OPTIONS=detect_stack_use_after_return=0` because this was problematic as well.

* disable flaky tests

kostasrim commented:
https://github.com/dragonflydb/dragonfly/actions/runs/10339261531 :green_circle: finally

chakaz reviewed (approved):
:shrug:

romange reviewed (approved):
so far sanitizers built did not provide as much benefits, only the maintenance cost

kostasrim commented:
> so far sanitizers built did not provide as much benefits, only the maintenance cost

I kinda agree. Kinda because when I introduced them they did find a few cases which I fixed
